example.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * This is a hypothetical implementation to understand
+ * what implementations may look like as we define
+ * a standard for video quality of experience metrics.
+ */
+
+// States
+var NOT_PLAYING = 0;
+var ATTEMPTING_TO_PLAY = 1;
+var PLAYING = 2;
+
+
+// Watch Time
+var watchTime = 0;
+var lastWatchTimeCheck;
+var watchTimeInterval;
+
+function updateWatchTime() {
+  watchTime = watchTime + (Date.now() - lastWatchTimeCheck);
+  lastWatchTimeCheck = Date.now();
+}
+
+player.addEventListener('playrequest', function(){
+  lastWatchTimeCheck = Date.now();
+  watchTimeInterval = window.setInterval(updateWatchTime, 250);
+});
+
+function stopWatchTimeInterval() {
+  window.clearInterval(watchTimeInterval);
+  updateWatchTime();
+}
+
+player.addEventListener('pause', stopWatchTimeInterval);
+player.addEventListener('end', stopWatchTimeInterval);
```

Comment:
Time spent buffering after seek still counts in watch time, right? If so, this is good, but if not, a seek should stop it, too.